clang/test/Driver/clang-offload-bundler-bc-archive-support-linux.cpp

```diff
@@ -0,0 +1,121 @@
+// Target "host-x86_64-unknown-linux-gnu" only works on Linux
+// REQUIRES: system-linux
+
+// Ensure that bundled BC files in archives can work with:
+// TEST1:  clang-offload-bundler -list
+// TEST2:  clang-offload-bundler -check-section
+// TEST3:  clang-offload-bundler -unbundle             with single target
+// TEST4:  clang-offload-bundler -unbundle             with multiple targets
+//
+// In all these tests also ensure functionality with bundled object files still
+// works correctly.
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Make bundled object with targets:
+// sycl-spir64-unknown-unknown
+// host-x86_64-unknown-linux-gnu
+// RUN: %clangxx -fsycl -c %s -o %t_bundled.o
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Make three distinct BC files
+// RUN: %clangxx -fsycl -fsycl-device-only -DTYPE1 %s -o %t1.bc
+// RUN: %clangxx -fsycl -fsycl-device-only -DTYPE2 %s -o %t2.bc
+// RUN: %clangxx -fsycl -fsycl-device-only -DTYPE3 %s -o %t3.bc
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Bundle BC files to different targets:
+// host-spir64-unknown-unknown
+// host-spir64_gen
+// host-spir64_x86_64
+// RUN: clang-offload-bundler -type=bc -targets=host-spir64-unknown-unknown -input=%t1.bc  -output=%t1_bundled.bc
+// RUN: clang-offload-bundler -type=bc -targets=host-spir64_gen             -input=%t2.bc  -output=%t2_bundled.bc
+// RUN: clang-offload-bundler -type=bc -targets=host-spir64_x86_64          -input=%t3.bc  -output=%t3_bundled.bc
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Make archive with bundled BC and o files
+// RUN: rm -f %t_bundled.a
+// RUN: llvm-ar cr %t_bundled.a %t1_bundled.bc %t2_bundled.bc %t3_bundled.bc %t_bundled.o
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// TEST1
+// Check that -list with various archive types can find all targets
+// RUN: clang-offload-bundler -list -type=ao  -input=%t_bundled.a | FileCheck --check-prefixes=CHECK-LIST %s
+// RUN: clang-offload-bundler -list -type=aoo -input=%t_bundled.a | FileCheck --check-prefixes=CHECK-LIST %s
+
+// CHECK-LIST-DAG: sycl-spir64-unknown-unknown
+// CHECK-LIST-DAG: host-x86_64-unknown-linux-gnu
+// CHECK-LIST-DAG: host-spir64-unknown-unknown
+// CHECK-LIST-DAG: host-spir64_gen
+// CHECK-LIST-DAG: host-spir64_x86_64
+
+// RUN: clang-offload-bundler -list -type=ao  -input=%t_bundled.a | wc -l | FileCheck --check-prefixes=CHECK-LIST-LENGTH %s

```

Comment:
Instead of running this twice, you can save the earlier output and run the FileCheck and 'wc -l' check on that output.
Same for the next command as well.